core/chains/evm/logpoller/log_poller_test.go

```diff
@@ -667,6 +685,127 @@ func TestLogPoller_LoadFilters(t *testing.T) {
 	assert.True(t, filter3.contains(&filter))
 }
 
+func TestLogPoller_convertLogs(t *testing.T) {
+	t.Parallel()
+	ctx := testutils.Context(t)
+	th := SetupTH(t, 2, 3, 2)
+
+	addresses := []common.Address{th.EmitterAddress1, th.EmitterAddress2}
+	topics := []common.Hash{EmitterABI.Events["Log1"].ID, EmitterABI.Events["Log2"].ID}
+
+	err := th.LogPoller.RegisterFilter(Filter{"convertLogs", topics, addresses})
```

Comment:
I should have renamed this test--Test_convertLogs is misleading.  Originally I was planning to have  one test for `lp.convertLogs()` and a separate one for `lp.blocksFromLogs()`, but after realizing there is no way to mock an eth client from logpoller (due to circular dependencies), I ended up deciding the best option was to just have a single test for both.  Also, since it's basically an end-to-end test of how blockpoller handles block timestamps as the chain progresses, I think I should move it into `logpoller/integration_test.go`.

I've converted `convertLogs()` back into a pure function, and added a 2 parameter version of it in `helper_test.go` that fills in the logger and chain id params for convenience.  I can also add a simple standalone `convertLogs()` test, but I think the end-to-end test is still useful to have.

Reply:
blocksFromLogs is such a thin wrapper around GetBlocksRange I probably wouldn't even worry about it (has its own test) and just add maybe a small assertion to the existing integration test + pure convertLogs tests